File: appMain/src/main/java/com/cartlc/tracker/act/NoteListAdapter.java
package com.cartlc.tracker.act;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.cartlc.tracker.R;
import com.cartlc.tracker.data.DataNote;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by dug on 5/12/17.
 */

public class NoteListAdapter extends RecyclerView.Adapter<NoteListAdapter.CustomViewHolder> {

    protected class CustomViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.label) TextView label;
        @BindView(R.id.value) TextView value;

        public CustomViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }

    final protected Context        mContext;
    final protected LayoutInflater mLayoutInflater;
    protected       List<DataNote> mItems;

    public NoteListAdapter(Context context) {
        mContext = context;
        mLayoutInflater = LayoutInflater.from(mContext);
    }

    @Override
    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = mLayoutInflater.inflate(R.layout.entry_item_note, parent, false);
        return new CustomViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final CustomViewHolder holder, final int position) {
        final DataNote item = mItems.get(position);
        holder.label.setText(item.name);
        holder.value.setText(item.value);
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }

    public void setItems(List<DataNote> items) {
        mItems = items;
        notifyDataSetChanged();
    }

}

File: appMain/src/main/java/com/cartlc/tracker/ui/list/NoteListAdapter.kt
/**
 * Copyright 2018, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.ui.list

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

import com.cartlc.tracker.R
import com.cartlc.tracker.model.data.DataNote
import kotlinx.android.synthetic.main.entry_item_note.view.*

/**
 * Created by dug on 5/12/17.
 */

class NoteListAdapter(mContext: Context) : RecyclerView.Adapter<NoteListAdapter.CustomViewHolder>() {
    protected val mLayoutInflater: LayoutInflater
    protected var mItems: List<DataNote> = mutableListOf()

    inner class CustomViewHolder(val view: View) : RecyclerView.ViewHolder(view) {

        fun bind(item: DataNote) {
            with (view) {
                label!!.text = item.name
                value!!.text = item.value
            }
        }
    }

    init {
        mLayoutInflater = LayoutInflater.from(mContext)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomViewHolder {
        val view = mLayoutInflater.inflate(R.layout.entry_item_note, parent, false)
        return CustomViewHolder(view)
    }

    override fun onBindViewHolder(holder: CustomViewHolder, position: Int) {
        holder.bind(mItems[position])
    }

    override fun getItemCount(): Int {
        return mItems.size
    }

    fun setItems(items: List<DataNote>) {
        mItems = items
        notifyDataSetChanged()
    }

}
